server/src/main/java/org/opensearch/tasks/SearchTaskResourceOperationsListener.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tasks;
+
+import org.opensearch.action.search.SearchPhaseContext;
+import org.opensearch.action.search.SearchRequestContext;
+import org.opensearch.action.search.SearchRequestOperationsListener;
+
+/**
+ * SearchTaskResourceOperationsListener subscriber for operations on search tasks resource usages.
+ * Listener ensures to refreshResourceStats on request end capturing the search task resource usage
+ * upon request completion.
+ *
+ * @PublicApi
+ */
+public final class SearchTaskResourceOperationsListener extends SearchRequestOperationsListener {
```

Comment:
```suggestion
public final class SearchTaskRequestOperationsListener extends SearchRequestOperationsListener {
```

Reply:
(suggestion, I think it reflects that this is the request listener, not task listener)